src/idl_gen_cpp.cpp

```diff
@@ -233,6 +233,28 @@ class CppGenerator : public BaseGenerator {
     return keywords_.find(name) == keywords_.end() ? name : name + "_";
   }
 
+  std::string Name(const FieldDef &field) const {
```

Comment:
Can we make this a template specialization instead of a virtual overload? I was confused at first where we were using this new function, but then realized FieldDef is a Definition.

Reply:
not sure what your asking for here - this is the cleanest most surgical way that I could find of modifying just fields, and not other things that are definitions

Reply:
(unless you want Koenig/ADL lookup via an injected friend function within FieldDef? but that would be worse obfuscation in my opinion)

Reply:
Something like 

```
  template<typename T>
  std::string Name(const T &def) const {
    return EscapeKeyword(def.name);
  }
```

and then you specialize it

```
  template<>
  std::string Name(const FieldDef &def) const {
    // your code
  }
```

Then we can avoid virtual overloads.

Reply:
the existing code already overloads EnumVal and Definition - note that these overloads are compile-time static overloads, not virtual; if I put a template <typename T> in there, it risks matching types other than Definition that aren't intended

Reply:
Yeah, i guess all cases are compile-timed known, so it should be fine.